ndc-client/src/apis/default_api.rs

```diff
@@ -34,6 +34,23 @@ fn inject_trace_context(builder: RequestBuilder) -> RequestBuilder {
     req_builder
 }
 
+fn append_path(url: &reqwest::Url, path: &str) -> Result<reqwest::Url, ConnectorURLError> {
+    if url.path_segments().map_or(false, |mut s|
+        // It is safe to unwrap here as according to documentation of Url::path_segments()
+        // > When Some is returned, the iterator always contains at least one string (which may be empty).
+        s.next_back().unwrap() != "")
+    {
+        let mut url = url.clone();
+        // No trailing slash, add it
+        url.path_segments_mut()
+            .map_err(|_| ConnectorURLError::URLCannotBeABase())?
+            .push("");
```

Comment:
Why not just push `path` here and skip the extra call to `join` (and presumably allocation)?